servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/FlushStrategyOnServerTest.java

```diff
@@ -112,19 +125,36 @@ public void flush(final ChannelHandlerContext ctx) {
             return succeeded(resp);
         };
         DefaultHttpExecutionContext httpExecutionContext =
-                new DefaultHttpExecutionContext(DEFAULT_ALLOCATOR, ioExecutor, executor, executionStrategy);
-        ReadOnlyHttpServerConfig config = new HttpServerConfig().asReadOnly();
+                new DefaultHttpExecutionContext(DEFAULT_ALLOCATOR, ioExecutor, executor, param.executionStrategy);
+
+        final ReadOnlyHttpServerConfig config = new HttpServerConfig().asReadOnly();
         serverConnection = initChannel(channel, httpExecutionContext, config,
-                new TcpServerChannelInitializer(config.tcpConfig()), service, true,
+                new TcpServerChannelInitializer(config.tcpConfig()) {
+                    @Override
+                    public void init(final Channel channel) {
+                        super.init(channel);
+                        channel.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                            @Override
+                            public void userEventTriggered(final ChannelHandlerContext ctx,
+                                                           final Object evt) throws Exception {
+                                if (evt == CloseHandler.ProtocolPayloadEndEvent.OUTBOUND) {
+                                    // Mute payload boundary events for this test with repeated read
```

Comment:
Can you explain the motivation for this change?

`ProtocolPayloadEndEvent.OUTBOUND` is for write but the comment here suggests it is due to "repeated reads".

In case you end up removing this change, we can make  the class `ProtocolPayloadEndEvent` package-private.

Reply:
The repeated read, single perpetual write mechanism is root cause for some of the items you had remarks on. If the protocol event were to be propagated here the `WriteStreamSubscriber` would be terminated which effectively stops requesting data from the source. Some tests send multiple requests on the same connection. 

It's the same reason we conditionally emit the event from the `RequestResponseCloseHandler` (cfr. `isClient` mentioned above) and the cause for piling up of listeners on server write futures until channel close.

If I were to split out an H2 variant from the `UnsupportedProtocolCloseHandler`, this would go away.

Reply:
>> If I were to split out an H2 variant from the UnsupportedProtocolCloseHandler, this would go away.

Ok I think we should split out an H2 variant and not emit the event from `UnsupportedProtocolCloseHandler `.